### DISCUSSION TITLE
How do we get field values from different versions of a page

### DISCUSSION THREAD

**Discussion by ahamedfazil** [pnp/pnpcore#1329] [Q&A]
Hi,

I want to get the field values for a page from different version. For example, I have a page ABC.aspx and it has 3 major version and 2 minor version. Here I would like to loop through each version and get a value of a specific field lets say "Author" field.

I followed below docs and samples,

[getting-the-file-version-of-a-list-item-version](https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#getting-the-file-version-of-a-list-item-version)
[updating-the-page-author-editor-created-and-modified-system-properties](https://pnp.github.io/pnpcore/using-the-sdk/pages-intro.html#updating-the-page-author-editor-created-and-modified-system-properties)

Here is my code snippet I tried,

![image](https://github.com/pnp/pnpcore/assets/21005625/2ad9c497-7d2b-4890-94b5-ca107348a5b7)

Thanks,

**Comment by jansenbe** [comment]
@ahamedfazil : the pages library is a form of a document library, so you'll need to get previous file versions for it like explained in the first link and then get the needed information from the returned `IFileVersion`. We do not have an implementation that loads the page versions into the page API code, so you'll not have the advanced page features for versions of a page but rather you have to do this via looking at `AllProperties` attribute of the returned `IFileVersion`

**Reply by ahamedfazil** [reply]
Thanks @jansenbe, it makes sense. I can get the field values from different versions by calling ListItem.
